tests_new/integration_tests/modules/s3_datasets/test_s3_tables_columns.py

```diff
@@ -7,35 +8,81 @@
+def test_sync_dataset_table_columns(client1, tables_fixture_name, request):
+    tables = request.getfixturevalue(tables_fixture_name)
+    response = sync_dataset_table_columns(client1, tables[0].tableUri)
+    assert_that(response.count).is_equal_to(3)
+    assert_that(response.nodes[0].name).is_equal_to('column1')
+    assert_that(response.nodes[0].typeName).is_equal_to('int')
 
 
-def test_sync_dataset_table_columns_unauthorized():
-    # TODO
-    pass
+@pytest.mark.parametrize(
+    'dataset_fixture_name,tables_fixture_name',
+    [('session_s3_dataset1', 'session_s3_dataset1_tables')],
+)
+def test_sync_dataset_table_columns_unauthorized(client2, dataset_fixture_name, tables_fixture_name, request):
+    dataset = request.getfixturevalue(dataset_fixture_name)
+    tables = request.getfixturevalue(tables_fixture_name)
+    table_uri = tables[0].tableUri
+    assert_that(sync_dataset_table_columns).raises(GqlError).when_called_with(client2, table_uri).contains(
+        'UnauthorizedOperation', 'UPDATE_DATASET_TABLE', dataset.datasetUri
+    )
 
 
-def test_update_dataset_table_column():
-    # TODO
-    pass
+@pytest.mark.parametrize(*TABLES_FIXTURES_PARAMS)
+def test_list_dataset_table_columns(client1, tables_fixture_name, request):
+    tables = request.getfixturevalue(tables_fixture_name)
+    table_uri = tables[0].tableUri
+    response = list_dataset_table_columns(client1, table_uri)
+    assert_that(response.count).is_equal_to(3)
+    assert_that(response.nodes[0].name).is_equal_to('column1')
+    assert_that(response.nodes[0].columnUri).is_not_none()
 
 
-def test_update_dataset_table_column_unauthorized():
-    # TODO
-    pass
+@pytest.mark.parametrize(*TABLES_CONFIDENTIALITY_FIXTURES_PARAMS)
+def test_list_dataset_table_columns_by_confidentiality(client2, tables_fixture_name, confidentiality, request):
+    tables = request.getfixturevalue(tables_fixture_name)
+    table_uri = tables[0].tableUri
+    if confidentiality in ['Unclassified']:
+        response = list_dataset_table_columns(client2, table_uri)
+        assert_that(response.count).is_equal_to(3)
+    else:
+        assert_that(list_dataset_table_columns).raises(GqlError).when_called_with(client2, table_uri).contains(
+            'UnauthorizedOperation', 'LIST_DATASET_TABLE_COLUMNS'
+        )
 
 
-def test_list_dataset_table_columns():
-    # TODO
-    pass
+@pytest.mark.parametrize(*TABLES_FIXTURES_PARAMS)
+def test_update_dataset_table_column(client1, tables_fixture_name, request, session_id):
+    tables = request.getfixturevalue(tables_fixture_name)
+    table_uri = tables[0].tableUri
+    columns = list_dataset_table_columns(client1, table_uri)
+    column_uri = columns.nodes[0].columnUri
+    response = update_dataset_table_column(
+        client1, column_uri, {'description': f'{session_id} new updated description'}
+    )
+    assert_that(response.description).is_equal_to(f'{session_id} new updated description')
```

Comment:
nit: I would minimize risk of accidental typos, if we are to change this
```
new_desc = f'{session_id} new updated description'
response = update_dataset_table_column(
        client1, column_uri, {'description': new_desc}
    )
assert_that(response.description).is_equal_to(new_desc)
```

Reply:
Done!